Lib/multiprocessing/shared_memory.py

```diff
@@ -198,7 +200,14 @@ def buf(self):
     @property
     def name(self):
         "Unique name that identifies the shared memory block."
-        return self._name
+        reported_name = self._name
+        if _USE_POSIX and self._prepend_leading_slash:
+            try:
+                if self._name[0] == "/":
+                    reported_name = self._name[1:]
+            except Exception:
```

Comment:
My concern was the possibility that ``self._name`` could be set to something other than a ``str`` (especially if it were ``None``).  Granted, I have not tested using ``bytes`` for the name, but it did cross my mind that it might work, in which case ``self._name[0] == "/"`` would still work but ``self._name.startswith("/")`` would complain that it was not passed ``b"/"``.

Reply:
The C module only accepts strings so you'll fail in __init__ (where `_name` is always set). Don't worry about nasty things users may do with it (not your responsability)

Reply:
Ah, good.  I have removed the try-except and changed to use ``str.startswith`` in commit 6f513bd6d5ebc675d57b5cdddd2e28b3f0548e7c.

I almost can't help but think about nasty things users may do.... the horror...  =)